mne_bids/mne_bids.py

```diff
@@ -424,60 +506,64 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
 
     make_dataset_description(output_path, name=" ",
                              verbose=verbose)
-    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind, verbose)
+    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind,
+                  eeg_reference, verbose)
     _channels_tsv(raw, channels_fname, verbose)
 
     events = _read_events(events_data, raw)
     if len(events) > 0:
-        _events_tsv(events, raw, events_tsv_fname, event_id, verbose)
+        _events_tsv(events, raw, events_tsv_fname, event_id, event_type,
+                    verbose)
 
+    # Before writing the neural data, cover some special cases:
     # for FIF, we need to re-save the file to fix the file pointer
     # for files with multiple parts
-    if ext in ['.fif', '.gz']:
+    if ext in ['.fif', '.fif.gz']:
         raw.save(raw_file_bids, overwrite=overwrite)
+
+    # If NOT fif, check if we got a multifile format
     else:
-        if os.path.exists(raw_file_bids):
-            if overwrite:
-                os.remove(raw_file_bids)
-                sh.copyfile(raw_file, raw_file_bids)
+        # Cover BrainVision multifile format of .vhdr, .eeg, .vmrk
+        if ext in ['.eeg']:
+            # Get future file names
+            fname, ext = _parse_ext(raw_file_bids)
+            assert ext == '.eeg'
+            assert fname + ext == raw_file_bids
+            raw_file_bids_list = [fname + '.eeg',
+                                  fname + '.vhdr',
+                                  fname + '.vmrk']
+            # Get locations of current files
+            fname, ext = _parse_ext(raw_file)
+            assert ext == '.eeg'
+            assert fname + ext == raw_file
+            raw_file_list = [fname + '.eeg',
+                             fname + '.vhdr',
+                             fname + '.vmrk']
+        else:
+            # All other data are single file formats
+            raw_file_bids_list = [raw_file_bids]
+            raw_file_list = [raw_file]
+
+        # Now copy&move or overwrite the files for the new BIDS directory
+        for raw_file, raw_file_bids in zip(raw_file_list, raw_file_bids_list):
+            if os.path.exists(raw_file_bids):
+                if overwrite:
+                    os.remove(raw_file_bids)
```

Comment:
why is this necessary? Doesn't `sh.copyfile` overwrite the file?

Reply:
Yes, `sh.copyfile` overwrites ... sometimes we don't want that, that's why there is this clause I guess - I did not add it though, it's old